7d575701e377fcd88202af2 Mon Sep 17 00:00:00 2001
From: Samuel Verschelde <stormi-xcp@ylix.fr>
Date: Tue, 2 Aug 2022 11:16:43 +0200
Subject: [PATCH] Fix quicktest's -default-sr parameter

The value of `use_default_sr` is evaluated before it can be set to true
if the `-default-sr` parameter is provided to the command line.

Move the if test inside a function so that `use_default_sr` is only
evaluated when appropriate.

Signed-off-by: Samuel Verschelde <stormi-xcp@ylix.fr>
---
 ocaml/quicktest/qt_filter.ml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ocaml/quicktest/qt_filter.ml b/ocaml/quicktest/qt_filter.ml
index 14a154b96e..7f132624d8 100644
--- a/ocaml/quicktest/qt_filter.ml
+++ b/ocaml/quicktest/qt_filter.ml
@@ -172,15 +172,16 @@ module SR = struct
 
   let all_srs = lazy (list_srs_connected_to_localhost !A.rpc !session_id)
 
-  let all =
+  let all () =
     if !A.use_default_sr then
       let pool = Qt.get_pool !A.rpc !session_id in
       only
         (Client.Client.Pool.get_default_SR ~rpc:!A.rpc ~session_id:!session_id
            ~self:pool
         )
+        ()
     else
-      fun () -> Lazy.force all_srs
+      Lazy.force all_srs
 
   let random srs () =
     let srs = srs () in

